frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx

```diff
@@ -0,0 +1,205 @@
+        search: 'severity',
+        id: 'severity',
+        exportContent: (item) => item.severity,
+      },
+      {
+        header: t('Source'),
+        cell: (item: DiscoverdPolicyTableItem) => discoveredSourceCell(t, item.source),
+        sort: (a: DiscoverdPolicyTableItem, b: DiscoverdPolicyTableItem) =>
+          compareStrings(a.source?.type, b.source?.type),
+        search: (item: DiscoverdPolicyTableItem) => item.source?.type ?? '',
+        id: 'source',
+        exportContent: (item) => item?.source?.type ?? '-',
+      },
+      {
+        header: t('Violations'),
+        cell: (item: DiscoverdPolicyTableItem) => {
+          const { noncompliant, compliant, pending, unknown } = policyViolationSummary(item.policies)
+          const path = generatePath(NavigationPath.discoveredPolicyDetails, {
+            clusterName: item.policies[0].cluster ?? '',
+            apiGroup: item.policies[0].apigroup ?? '',
+            apiVersion: item.policies[0].apiversion ?? '',
+            kind: item.kind,
+            templateName: item.name,
+          })
+          if (noncompliant !== 0 || compliant !== 0 || pending != 0 || unknown !== 0) {
+            return (
+              <ClusterPolicyViolationIcons2
+                compliant={compliant}
+                compliantHref={path}
+                noncompliant={noncompliant}
+                violationHref={path}
+                pending={pending}
+                pendingHref={path}
+                unknown={unknown}
+              />
+            )
+          }
+          return '-'
+        },
+        sort: 'violations',
+        search: 'violations',
+        id: 'violations',
+        exportContent: (item: DiscoverdPolicyTableItem) => {
+          const { noncompliant, compliant, pending, unknown } = policyViolationSummary(item.policies)
+          return `${t('no violations: {{count}} cluster', { count: compliant })}, ${t('violations: {{count}} cluster', { count: noncompliant })}, ${t('pending: {{count}} cluster', { count: pending })}, ${t('unknown: {{count}} cluster', { count: unknown })}`
+        },
+      },
+    ],
+    [t]
+  )
+
+  const filters = useMemo<ITableFilter<DiscoverdPolicyTableItem>[]>(
+    () => [
+      {
+        id: 'violations',
+        label: 'Cluster violations',
+        options: [
+          {
+            label: t('No violations'),
+            value: 'no-violations',
+          },
+          {
+            label: t('Violations'),
+            value: 'violations',
+          },
+        ],
+        tableFilterFn: (selectedValues, item) => {
+          const { noncompliant, compliant } = policyViolationSummary(item.policies)
+
+          if (selectedValues.includes('no-violations') && noncompliant == 0) {
+            return true
+          } else if (selectedValues.includes('violations') && compliant > 0) return true
+          return false
+        },
+      },
+      {
+        id: 'kind',
+        label: t('Kind'),
+        options: [
```

Comment:
Let's make this dynamic. This will help when for when we add Gatekeeper constraints.

See this diff:

```diff
diff --git a/frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx b/frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx
index 0d200649..d284ea21 100644
--- a/frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx
+++ b/frontend/src/routes/Governance/discovered/DiscoveredPolicies.tsx
@@ -1,7 +1,14 @@
 /* Copyright Contributors to the Open Cluster Management project */
 import { EmptyState, EmptyStateIcon, PageSection, Spinner, Title } from '@patternfly/react-core'
 import { AcmTable } from '../../../ui-components/AcmTable'
-import { AcmEmptyState, compareStrings, ITableFilter, IAcmTableColumn } from '../../../ui-components'
+import {
+  AcmEmptyState,
+  compareStrings,
+  ITableFilter,
+  IAcmTableColumn,
+  TableFilterOption,
+  FilterOptionValueT,
+} from '../../../ui-components'
 import { DiscoverdPolicyTableItem, useFetchPolicies } from './useFetchPolicies'
 import { useTranslation } from '../../../lib/acm-i18next'
 import { useMemo } from 'react'
@@ -121,6 +128,21 @@ export default function DiscoveredPolicies() {
     [t]
   )
 
+  const kindsSet = new Set<string>()
+
+  for (const item of data || []) {
+    kindsSet.add(item.kind)
+  }
+
+  const kinds = [...kindsSet]
+  kinds.sort()
+
+  const kindOptions: TableFilterOption<FilterOptionValueT>[] = []
+
+  kinds.map((kind: string) => {
+    kindOptions.push({ label: kind, value: kind })
+  })
+
   const filters = useMemo<ITableFilter<DiscoverdPolicyTableItem>[]>(
     () => [
       {
@@ -148,11 +170,7 @@ export default function DiscoveredPolicies() {
       {
         id: 'kind',
         label: t('Kind'),
-        options: [
-          { label: t('Configuration policy'), value: 'ConfigurationPolicy' },
-          { label: t('Operator policy'), value: 'OperatorPolicy' },
-          { label: t('Certificate policy'), value: 'CertificatePolicy' },
-        ],
+        options: kindOptions,
         tableFilterFn: (selectedValues, item) => {
           return selectedValues.includes(item.kind)
         },
@@ -171,7 +189,7 @@ export default function DiscoveredPolicies() {
         },
       },
     ],
-    [t]
+    [t, kindOptions]
   )
 
   if (isFetching) {
@@ -186,6 +204,7 @@ export default function DiscoveredPolicies() {
       </PageSection>
     )
   }
+
   /* t('preview.discoveredPolicies') */
   return (
     <PageSection>
diff --git a/frontend/src/ui-components/AcmTable/AcmTable.tsx b/frontend/src/ui-components/AcmTable/AcmTable.tsx
index 587bd843..917e0a99 100644
--- a/frontend/src/ui-components/AcmTable/AcmTable.tsx
+++ b/frontend/src/ui-components/AcmTable/AcmTable.tsx
@@ -217,8 +217,8 @@ interface ITableItem<T> {
   [key: string]: unknown
 }
 
-type FilterOptionValueT = string
-type TableFilterOption<FilterOptionValueT> = { label: ReactNode; value: FilterOptionValueT }
+export type FilterOptionValueT = string
+export type TableFilterOption<FilterOptionValueT> = { label: ReactNode; value: FilterOptionValueT }
 export type TableFilterFn<T> = (selectedValues: string[], item: T) => boolean
 /**
  * Interface defining required params for table filtering property "filterItems"
```

Reply:
by the way why does all diff not work... I tried `git appy` then response with `error: corrupt patch at line 86`

Reply:
This way... We cannot use `t()` is that ok?